library/psa_crypto.c

```diff
@@ -3609,6 +3609,50 @@ psa_status_t psa_cipher_decrypt( mbedtls_svc_key_id_t key,
 /* AEAD */
 /****************************************************************/
 
+/* Helper function to get the base algorithm from its variants. */
+static psa_algorithm_t psa_aead_get_base_algorithm( psa_algorithm_t alg )
+{
+    return PSA_ALG_AEAD_WITH_DEFAULT_LENGTH_TAG( alg );
+}
+
+/* Helper function to perform common nonce length checks. */
+static psa_status_t psa_aead_check_nonce_length( psa_algorithm_t alg,
+                                                 size_t nonce_length )
+{
+    psa_algorithm_t base_alg = psa_aead_get_base_algorithm( alg );
```

Comment:
While at it, what about to make this function more robust by moving to a whitelist approach rather than a blacklist one:
```
if( base_alg == PSA_ALG_GCM )
{
    if( nonce_length != 0 )
        return( PSA_SUCCESS );
}
...
return( PSA_ERROR_NOT_SUPPORTED );
```